src/Infrastructure/DefinitionResolver.php

```diff
@@ -58,14 +58,15 @@ public static function resolveArray(DependencyResolverInterface $dependencyResol
      *
      * @return mixed
      */
-    public static function resolve(DependencyResolverInterface $dependencyResolver, $definition)
+    public static function resolve(ContainerInterface $container, ?ContainerInterface $referenceContainer, $definition)
     {
         if ($definition instanceof DefinitionInterface) {
+            $container = $referenceContainer !== null && $definition instanceof ReferenceInterface ? $referenceContainer : $container;
             /** @var mixed $definition */
-            $definition = $definition->resolve($dependencyResolver);
+            $definition = $definition->resolve($container);
```

Comment:
Don't good override exists variable `$container`. Suggestion:

```php
$definition = $definition->resolve(
	$referenceContainer !== null && $definition instanceof ReferenceInterface
		? $referenceContainer
		: $container
);
```

Reply:
Why not?

Reply:
bad practic